### PR TITLE
Add tests for reading numeric limits in parquet statistics

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/10453

## Rationale for this change

As we fill out parquet statistics features, it would be good to ensure they work well with numeric limits (aka the ranges of values of different types)

## What changes are included in this PR?

Add new tests for numeric limits

## Are these changes tested?

Only tests

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 1]
```diff
@@ -212,13 +218,13 @@ impl Test {
         let expected_null_counts = Arc::new(expected_null_counts) as ArrayRef;
         assert_eq!(
             &null_counts, &expected_null_counts,
-            "Mismatch with expected null counts"
+            "{column_name}: Mismatch with expected null counts"
```
Comment (comphead) [inline datafusion/core/tests/parquet/arrow_statistics.rs:221]:
```suggestion
            "{column_name}: Mismatch with expected null counts. Actual: {null_counts}. Expected: {expected_null_counts}"
```

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 2]
```diff
@@ -212,13 +218,13 @@ impl Test {
...
         );
 
         let row_counts = StatisticsConverter::row_counts(reader.metadata()).unwrap();
         assert_eq!(
             row_counts, expected_row_counts,
-            "Mismatch with expected row counts"
+            "{column_name}: Mismatch with expected row counts"
```
Comment (comphead) [inline datafusion/core/tests/parquet/arrow_statistics.rs:227]:
```suggestion
            "{column_name}: Mismatch with expected row counts. Actual: {row_counts}. Expected: {expected_row_counts}"
```

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

**Comment by alamb** [comment]
Thanks again for the review @comphead

**Pull request merged**